test/e2e/tls_test.go

```diff
@@ -65,14 +65,14 @@ func TestAntreaApiserverTLSConfig(t *testing.T) {
 		apiserver     int
 		apiserverStr  string
 	}{
-		{"ControllerApiserver", controllerPodName, controllerContainerName, apis.AntreaControllerAPIPort, "Controller"},
+		{"ControllerApiserver", agentPodName, agentContainerName, apis.AntreaControllerAPIPort, "Controller"},
 		{"AgentApiserver", agentPodName, agentContainerName, apis.AntreaAgentAPIPort, "Agent"},
 	}
 	for _, tc := range tests {
 		tc := tc
 		t.Run(tc.name, func(t *testing.T) {
 			t.Parallel()
-			data.checkTLS(t, tc.podName, tc.containerName, tc.apiserver, tc.apiserverStr)
+			data.checkTLS(t, tc.podName, tc.containerName, tc.apiserver, tc.apiserverStr, controllerPodIP)
```

Comment:
You need to get both agent and controller IP and passed it here, otherwise, both two cases will actually check controller API.

Reply:
@luolanzone so instead of passing the agentPodIP as well, can we add a condition that if the port is 10349 then we can directly use the localhostIP because we are running this command from the antrea-agent pod.
What's your suggestion?

Reply:
done